numpy/core/src/multiarray/ctors.c

```diff
@@ -784,6 +784,33 @@ PyArray_NewFromDescr_int(
 
 
     if (data == NULL) {
+        NPY_traverse_info fill_zero_info;
+        /* float errors do not matter and we do not release GIL */
+        NPY_ARRAYMETHOD_FLAGS zero_flags = PyArrayMethod_MINIMAL_FLAGS;
```

Comment:
```suggestion
        NPY_ARRAYMETHOD_FLAGS zero_flags;
```
No need here (not that it matters).  It's relevant for structured dtypes or chained functions only.